Sources/SwiftFormat/Rules/BeginDocumentationCommentWithOneLineSummary.swift

```diff
@@ -130,8 +130,14 @@ public final class BeginDocumentationCommentWithOneLineSummary: SyntaxLintRule {
         in: text.startIndex..<text.endIndex,
         scheme: NSLinguisticTagScheme.lexicalClass.rawValue,
         tokenRanges: &tokenRanges)
+      var isInsideQuotes = false
       let sentenceTerminatorIndices = tags.enumerated().filter {
-        $0.element == "SentenceTerminator"
+        if $0.element == "OpenQuote" {
+          isInsideQuotes = true
+        } else if $0.element == "CloseQuote" {
+          isInsideQuotes = false
```

Comment:
I was going to point out that this won't handle nested quotes correctly, but it looks like the Foundation API itself doesn't categorize them correctly either:

```swift
func tags(_ text: String) {
  var tokenRanges = [Range<String.Index>]()
  let tags = text.linguisticTags(in: text.startIndex..<text.endIndex, scheme: NSLinguisticTagScheme.lexicalClass.rawValue, tokenRanges: &tokenRanges)
  print(Array(tags))
}
tags("\"Hello 'world'\"")
// ["OpenQuote", "Interjection", "Whitespace", "OpenQuote", "Noun", "CloseQuote", "OpenQuote"]
```

So an attempt to increment/decrement a nested quote count won't work as intended. In other words, I wanted to point it out but you don't need to change anything.

Reply:
Yes, this is entirely dependent on the behavior of the Foundation API.
And since `linguisticTags` is a deprecated API, it's unlikely that we'll see any improvements to it 🤔 

The replacement, `NaturalLanguage`, works the same as `linguisticTags`, but I've migrated to it in case it's improved in the future.